src/librustc_data_structures/sync.rs

```diff
@@ -271,6 +322,8 @@ cfg_if! {
 
         pub use std::sync::atomic::{AtomicBool, AtomicUsize, AtomicU32, AtomicU64};
 
+        pub use crossbeam_utils::atomic::AtomicCell;
```

Comment:
I don't get why you define the struct `AtomicCell` above but then use it from the crossbeam_utils? Are they different? What is the purpose of including AtomicCell from crossbeam_utils here?

Reply:
In general, this code defines a unified set of structures related to multi-threadiness(?) **with a caveat** - the definitions are different for `parallel_compiler` cfg.

If you expand the omitted lines in the diff, you can see that the `pub struct AtomicCell<T: Copy>(Cell<T>);` (which is mainly a wrapper around `Cell<T>`, which isn't really atomic nor should it be in single-threaded case) is defined manually in the `#[cfg(not(parallel_compiler))]`  branch.
Otherwise, if we build a parallel compiler, we just re-export `crossbeam_utils::atomic::AtomicCell` instead.